front/lib/models/assistant/actions/retrieval.ts

```diff
@@ -185,18 +192,32 @@ AgentDataSourceConfiguration.init(
 AgentRetrievalConfiguration.hasOne(AgentConfiguration, {
   foreignKey: { name: "retrievalConfigurationId", allowNull: true }, // null = no retrieval action set for this Agent
 });
+AgentConfiguration.belongsTo(AgentRetrievalConfiguration, {
+  as: "retrievalConfiguration",
+  foreignKey: { name: "retrievalConfigurationId", allowNull: true },
+});
 
 // Retrieval config <> Data source config
 AgentRetrievalConfiguration.hasMany(AgentDataSourceConfiguration, {
+  as: "dataSourceConfigurations",
   foreignKey: { name: "retrievalConfigurationId", allowNull: false },
   onDelete: "CASCADE",
 });
+AgentDataSourceConfiguration.belongsTo(AgentRetrievalConfiguration, {
```

Comment:
Why do we need this?

Reply:
Same as before the `as` can be added in the `hasOne` and this one can be removed